modules/c++/math/source/Utilities.cpp

```diff
@@ -0,0 +1,47 @@
+/* =========================================================================
+ * This file is part of math-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2017, MDA Information Systems LLC
+ *
+ * math-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <import/sys.h>
+#include <math/Utilities.h>
+
+namespace math
+{
+size_t nChooseK(size_t n, size_t k)
+{
+    if (n < k)
+    {
+        throw except::Exception(Ctxt("n Choose k undefined for n < k"));
+    }
+
+    // Algorithm to compute n Choose k without using factorials found here:
+    // http://csharphelper.com/blog/2014/08/
+    //         calculate-the-binomial-coefficient-n-choose-k-efficiently-in-c
+    size_t coefficient = 1;
```

Comment:
If we're on a 32-bit system (which nowadays is really rare but is still possible), a `size_t` is pretty small.  I think if you use a `sys::Uint64_T` throughout, you'll be safer.

There is some case where you'll overflow that too and would rather use a double.  Assuming you're using this for relatively small values for what you're doing right now, I think it's fine to leave it as an integer though to get a little more precision... if we ever hit an overflow case in the future, we can revisit.  Maybe for now just put a comment in the header cautioning the caller that we are doing integer arithmetic so it could overflow.